### DISCUSSION TITLE
Unnable to relate two fields to the same model (Ambiguous relation detected)

### DISCUSSION THREAD

**Discussion by romanzy313** [prisma/prisma#19118] [Q&A]
My app needs to support messaging where there are always 2 users in each conversation. My schema is the following:
```
model User {
  id Int @id @default(autoincrement())
  Conversations Conversation[] @relation(references: [userAId, userBId])
}
model Conversation {
  id Int @id @default(autoincrement())

  userA   User @relation(fields: [userAId], references: [id])
  userAId Int

  userB   User @relation(fields: [userBId], references: [id])
  userBId Int

  message Message @relation(fields: [messageId], references: [id])
  messageId Int
}
```

This is the error I get:
```
Ambiguous relation detected. The fields `userA` and `userB` in model `Conversation` both refer to `User`
```
The error message says exactly what I want to do, I want to define two users in the conversation. I could use an array of users in conversation model but it would be impractical, as each conversation must have exactly two users. I found these discussions: [here](https://github.com/prisma/prisma/discussions/14892) and [here](https://github.com/prisma/prisma/discussions/14345) but the solution says that there should be two unique fields in User model, which I am trying to avoid. I want to keep my user model same as above. This arrangement is possible to do with raw sql. Is this possible in Prisma?
Thank you!

**Comment by jharrell** [comment]
Hi @romanzy313! I'm not sure if you're going to be able to get around two unique fields on `User`, but have you tried adding `name` to the `@relation` attribute? See docs [here](https://www.prisma.io/docs/concepts/components/prisma-schema/relations#disambiguating-relations)

**Reply by janpio** [reply]
To add to that, if you have two relations from Conversation to User, you also need two relations back from User to Conversation. You can not just add both fields as `references`.

**Reply by nurul3101** [reply]
@romanzy313 Please let us know in case you need any further assistance.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
